MetaMask/metamask-extension
# feat: bump `@metamask/queued-request controller` to `^2.0.0` to fix confirmation popup focus in request queue

## **Description**



[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/25497?quickstart=1)

## **Related issues**

See: https://github.com/MetaMask/core/pull/4457
See: https://github.com/MetaMask/metamask-extension/issues/25397

## **Manual testing steps**

1. Ensure request queue is toggled on
2. In Tab 1, Visit the test dapp
3. run `await window.ethereum.request({method: 'wallet_switchEthereumChain', params: [{"chainId": "0xaa36a7"}]})` in console to change the network to sepolia for this dapp
4. In Tab 2, Visit a different dapp
5. run `await window.ethereum.request({method: 'wallet_switchEthereumChain', params: [{"chainId": "1"}]})` in console to change the network to mainnet for this dapp
6. In Tab 1, click the Send Transaction button, but ignore the confirmation notification (leave it open)
7. In Tab 2, run `await window.ethereum.request({method: 'eth_requestAccounts'})`
8. The confirmation notification window should be focused and visible now, but still be displaying the pending transaction from before

## **Screenshots/Recordings**



### **Before**



### **After**

https://github.com/MetaMask/metamask-extension/assets/918701/48693083-fe60-402f-8798-c32b5db541b6

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
